src/manifest.c

```diff
@@ -548,6 +536,16 @@ static int retrieve_manifests(int current, int version, char *component, struct
 		goto out;
 	}
 
+	/* Only signing the MoM manifest*/
+	if (strncmp(component, "MoM", 3) == 0) {
+		free(filename);
+		string_or_die(&filename, "%s/%i/Manifest.MoM", state_dir, version);
+		if (verify_signatures && !download_and_verify_signature(url, filename)) {
```

Comment:
Can the download if .sig not present be buried inside download_and_verify_signature()?  That matches how the manifest fetching happens...either fetch from disk else remote, but fetch either way.  Then it's maybe just verify_signature() with "download signature iff not already present" is abstracted inside the verifier?